android/src/main/java/com/applovin/reactnative/AppLovinMAXModule.java

```diff
@@ -1897,6 +1903,39 @@ private WritableMap createNetworkResponseInfo(final MaxNetworkResponseInfo respo
         return networkResponseObject;
     }
 
+    // Lifecycle Events
+
+    @Override
+    public void onHostResume() { }
+
+    @Override
+    public void onHostPause() { }
+
+    @Override
+    public void onHostDestroy()
+    {
+        // Converts mAdViews to a list since destroyAdView will modify mAdViews
+        List<MaxAdView> adViews = new ArrayList<MaxAdView>( mAdViews.values() );
+        for ( Iterator<MaxAdView> iterator = adViews.iterator(); iterator.hasNext(); )
+        {
+            MaxAdView adView = iterator.next();
+            iterator.remove();
+            destroyAdView( adView.getAdUnitId(), adView.getAdFormat() );
+        }
+
```

Comment:
Can we do something like this? Little bit more concise:
```suggestion
        // Make copy because `destroyAdView()` will remove from `mAdViews`
        List<MaxAdView> adViews = new ArrayList<>( mAdViews.values() );
        for ( MaxAdView adView : adViews )
        {
            destroyAdView( adView.getAdUnitId(), adView.getAdFormat() );
        }

```